File: backend/src/models/schemas/hazop.py
import typing
import datetime
import pydantic
from src.models.schemas.location import FieldLocationResponse

class DocumentMonthlyBatchCreate(pydantic.BaseModel):
    doc_type: str
    reporting_year: int
    reporting_month: int
    field: typing.Optional[str] = None
    mode: str = "append"
    items: list[dict[str, typing.Any]]

    class Config:
        allow_population_by_field_name = True
        orm_mode = True

class HazopResponse(pydantic.BaseModel):
    id: typing.Any
    upload_batch_id: typing.Any
    owner_account_id: typing.Optional[typing.Any]
    reporting_year: int
    reporting_month: int
    field: typing.Optional[str]
    field_location: typing.Optional[FieldLocationResponse] = None
    node_no: typing.Optional[str]
    rec_no: typing.Optional[str]
    node: typing.Optional[str]
    deviation: typing.Optional[str]
    possible_cause: typing.Optional[str]
    consequences: typing.Optional[str]
    safeguard: typing.Optional[str]
    recommendation: typing.Optional[str]
    likelihood: typing.Optional[str]
    severity: typing.Optional[str]
    risk: typing.Optional[str]
    responsibility_pic: typing.Optional[str]
    type: typing.Optional[str]
    target_date: typing.Optional[datetime.date]
    response_progress: typing.Optional[str]
    category: typing.Optional[str]
    sub_category: typing.Optional[str]
    status: typing.Optional[str]
    evidence: typing.Optional[str]
    completion_date: typing.Optional[datetime.date]
    sme: typing.Optional[str]
    created_at: datetime.datetime
    updated_at: typing.Optional[datetime.datetime]

    class Config:
        orm_mode = True

class HazopHistoryResponse(pydantic.BaseModel):
    upload_batch_id: typing.Any
    reporting_year: int
    reporting_month: int
    field: typing.Optional[str]
    upload_date: datetime.datetime
    record_count: int
    
    class Config:
        orm_mode = True

File: frontend/src/types/hazop.ts
export interface DocumentMonthlyBatchCreate {
	doc_type: string;
	reporting_year: number;
	reporting_month: number;
	mode?: "append" | "overwrite";
	items: Record<string, any>[];
}

export interface HazopHistoryResponse {
	upload_batch_id: string;
	reporting_year: number;
	reporting_month: number;
	upload_date: string;
	record_count: number;
}

export interface HazopDataResponse {
	id: string;
	upload_batch_id: string;
	owner_account_id: string | null;
	reporting_year: number;
	reporting_month: number;
	node_no: string | null;
	rec_no: string | null;
	node: string | null;
	deviation: string | null;
	possible_cause: string | null;
	consequences: string | null;
	safeguard: string | null;
	recommendation: string | null;
	likelihood: string | null;
	severity: string | null;
	risk: string | null;
	responsibility_pic: string | null;
	type: string | null;
	target_date: string | null;
	response_progress: string | null;
	category: string | null;
	sub_category: string | null;
	status: string | null;
	evidence: string | null;
	completion_date: string | null;
	sme: string | null;
	created_at: string;
	updated_at: string | null;
}
